src/methods/clipping/clipping_processor.jl

```diff
@@ -0,0 +1,286 @@
+# # This file contains the shared helper functions for the other polygon clipping
+# # functionalities.
+
+"""
+    _build_a_list(
+        poly_a, poly_b,
+        edges_a, edges_b, 
+        intr_list, a_idx_list, b_idx_list, 
+        alpha_a_list, alpha_b_list, 
+        a_list, b_list
+    )::intr_list, a_idx_list, b_idx_list, alpha_a_list, alpha_b_list
+
+    This function take in two polygons, the lists of their edges, and some lists containing
+    information about the nature of their intersection points. From that, it creates an 
+    array to represent polygon_a, and this array contains intersection points. However, after
+    calling this function, a_list is not fully formed because the neighboring indicies of the
+    intersection points in b_list still need to be updated. We will have fully formed a_list
+    after calling _build_ab_list. Also at this point we still have not update the entry and
+    exit flags for a_list. The variables poly_a and poly_b are the original polygons that the
+    user passed in. The variables edges_a and edges_b are lists of the edges of each polygon, 
+    in the form of their cartesian points. Then intr_list is a list of the intersection points
+    in their cartesion point form. Then a_idx_list is a list of the indicies of where in a_list
+    an intersection point lies. The value at index i of a_idx_list is the location in a_list where
+    the ith point of intr_list lies. Then b_idx_list indicates which EDGE of poylgon b an intersection
+    point lies. Then alpha_a_list and alpha_b_list are vectors of scalar values which help order
+    the intersection points in the case that they lie on the same edge of a polygon. Finally,
+    a_list is an empty array of PolyNode objects that gets filled out when this function is run.
+    So is b_list. 
+"""
+function _build_a_list(edges_a, edges_b,
+                      intr_list, a_idx_list, b_idx_list, alpha_a_list, alpha_b_list, a_list, b_list)
+    # Find intersection points and adds them to a_list
+    # "counter" is used to index all inter-related lists
+    counter = 1
+    # "acount" is used to index a_list
+    acount = 1
+    for ii in eachindex(edges_a)
+        # Add the first point of the edge to the list of points in a_list
+        if acount <= length(a_list)
+            a_list[acount] = PolyNode(ii, false, 0, false, 0)
+        else
+            push!(a_list, PolyNode(ii, false, 0, false, 0))
+        end
+        acount = acount + 1
+
+        # Keep track of current position in inter-related lists
+        # before finding new intersection points on our new edge
+        prev_counter = counter
+
+        for jj in eachindex(edges_b)
+
+            # Add the first point of the edge to b_list
+            if ii == 1 
```

Comment:
I think we should just build the entire `b_list` in the `_build_b_list` function rather than making some of it here